### PR TITLE
Feat(refs: T32135): new responsive styling header nav

### PR DESCRIPTION
**Pull request by elmasdemos**
**Ticket:** https://yaits.demos-deutschland.de/T32135

### Description: 

- 2 different navs are existing now:
- one for desktop version >= 992px and one for mobile version with a hamburger menu
- if hamburger menu open you see a close button for closing the menu
- both of the nav menus have different links
- before login the login button is `Anmelden`
- after login the login button is a user icon in the desktop version
- desktop version has a dropdown on title and about description left besides the logo
- on mobile version its in the nav menu the flyout included
- the width of the flyout is fix by 400px
- at the breakpoint 576px the menu flyout takes the fullscreen and you see the logo in the flyout
- Meine Daten` changes to `Mein Profil`
- profile setting and login and logout links get their own icons, so 3 new svgs are introduced
- after opened the hamburger menu the background get darkened
- therefore I used 1400vh because in some pages you can scroll down the page and in the dark background got a cut
- the main font-size changes to 15px, font-family stay open sans,
-  the smaller font-size is 14px for the role of the user and for the orga section name

### Linked PRs

- https://github.com/demos-europe/demosplan-project-diplanbau/pull/106

### PR Checklist

- [ ] Link all relevant tickets
- [x] Move the tickets on the board accordingly

### CODE REVIEW

**Review by muellerdemos — CHANGES_REQUESTED** [review]
Unfortunately the license headers are now duplicated (well there is a small change with the organisation name). However I think we should find a better way of solving this instead of manually adjusting the license headers in 596 files

File: client/js/InitVue.js [hunk 1]
```diff
@@ -11,7 +11,8 @@ import {
   DpAccordion,
   DpObscure,
   dpValidateMultiselectDirective,
-  Tooltip
+  Tooltip,
+  DpFlyout
```
Comment (muellerdemos) [inline client/js/InitVue.js:15]: nitpick: alphabetical sort order

File: client/js/InitVue.js [hunk 2]
```diff
@@ -24,6 +25,7 @@ import { initStore } from '@DpJs/store/core/initStore'
 import { loadLibs } from '@DpJs/lib/core/loadLibs'
 import loadSentry from './loadSentry'
 import NotificationStoreAdapter from '@DpJs/store/core/NotificationStoreAdapter'
+import NavButton from './components/button/NavButton'
```
Comment (muellerdemos) [inline client/js/InitVue.js:28]: I'm not sure that the component name represents its purpose. I think this is more likely a `ToggleMenuButton` from my perspective. However `NavButton` kind of suggests that the button is used to navigate but actually it toggles the navigation menu.

**Review by muellerdemos — CHANGES_REQUESTED** [review]

File: client/js/components/button/NavButton.vue [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<template>
+  <!--The attribute data-responsive-menu-helper is used as a hook to append an EventListener in GlobalEventlisteners.js-->
+  <button
+    type="button"
+    class="btn btn--blank font-size-medium"
+    aria-haspopup="true"
+    aria-expanded="false"
+    data-responsive-menu-helper>
+    <a class="nav-toggle">
```
Comment (spiess-demos) [inline client/js/components/button/NavButton.vue:9]: I guess placing an `a` element within a `button` element is invalid Html.
Also, the styles for `.nav-toggle` are located inside the project overrides - i'd suggest to also place them in the core, maybe in `scss/components/_<component-name>.scss`.

File: client/js/components/button/NavButton.vue [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      <span class="bar"></span>
+      <span class="bar"></span>
+      <span class="bar"></span>
```
Comment (spiess-demos) [inline client/js/components/button/NavButton.vue:12]:
```suggestion
      <span class="bar" />
      <span class="bar" />
      <span class="bar" />
```

File: client/js/components/button/NavButton.vue [hunk 5]
```diff
@@ -0,0 +1,23 @@
+<template>
+  <!--The attribute data-responsive-menu-helper is used as a hook to append an EventListener in GlobalEventlisteners.js-->
+  <button
+    type="button"
+    class="btn btn--blank font-size-medium"
+    aria-haspopup="true"
+    aria-expanded="false"
```
Comment (spiess-demos) [inline client/js/components/button/NavButton.vue:7]: There is no accessible label for the button element. Please use `aria-label` with an appropriate value.

Comment (elmasdemos) [reply client/js/components/button/NavButton.vue]: done https://github.com/demos-europe/demosplan-core/pull/1389/commits/692686822ba44f7c25e4e7bb57ddbbcf42bcf6af

File: client/js/components/button/NavButton.vue (hunk 3 above)
Comment (elmasdemos) [reply client/js/components/button/NavButton.vue]: done https://github.com/demos-europe/demosplan-core/pull/1389/commits/692686822ba44f7c25e4e7bb57ddbbcf42bcf6af

File: client/js/lib/core/libs/GlobalEventListener.js [hunk 6]
```diff
@@ -27,6 +27,8 @@ export default function initGlobalEventListener () {
   if (responsiveMenuHelper) {
     responsiveMenuHelper.addEventListener('click', function (event) {
       event.preventDefault()
+      let body = document.querySelector('body')
```
[the remark this answers is not in the document]
Comment (elmasdemos) [reply client/js/lib/core/libs/GlobalEventListener.js]: done https://github.com/demos-europe/demosplan-core/pull/1389/commits/692686822ba44f7c25e4e7bb57ddbbcf42bcf6af

File: client/js/InitVue.js (hunk 2 above)
Comment (elmasdemos) [reply client/js/InitVue.js]: i took `HamburgerMenuButton` as name. done here https://github.com/demos-europe/demosplan-core/pull/1389/commits/692686822ba44f7c25e4e7bb57ddbbcf42bcf6af

File: client/js/InitVue.js [hunk 7]
```diff
@@ -56,7 +58,7 @@ function initialize (components = {}, storeModules = {}, apiStoreModules = [], p
        * DpAccordion is registered globally here, because we need it for the sidemenu in sidemenu.html.twig and can't
        * register it locally there (special knp menu renderer, see https://github.com/KnpLabs/KnpMenu).
        */
-      components: { ...components, DpAccordion, NotifyContainer },
+      components: { ...components, DpAccordion, NotifyContainer, NavButton, DpFlyout },
```
[the remark this answers is not in the document]
Comment (elmasdemos) [reply client/js/InitVue.js]: done https://github.com/demos-europe/demosplan-core/pull/1389/commits/692686822ba44f7c25e4e7bb57ddbbcf42bcf6af

File: client/js/InitVue.js (hunk 1 above)
Comment (elmasdemos) [reply client/js/InitVue.js]: done https://github.com/demos-europe/demosplan-core/pull/1389/commits/692686822ba44f7c25e4e7bb57ddbbcf42bcf6af

**Review by muellerdemos — DISMISSED** [review]
Limited to my concerns: all concerns were addressed and this lgtm (code review)

File: client/js/components/button/HamburgerMenuButton.vue [hunk 8]
```diff
@@ -0,0 +1,24 @@
+<template>
+  <!--The attribute data-responsive-menu-helper is used as a hook to append an EventListener in GlobalEventlisteners.js-->
+  <button
+    type="button"
+    class="btn btn--blank font-size-medium"
+    aria-haspopup="true"
+    aria-expanded="false"
+    :aria-label="Translator.trans('menu.nav.toggle')"
+    data-responsive-menu-helper>
+    <a class="nav-toggle">
```
Comment (spiess-demos) [inline client/js/components/button/HamburgerMenuButton.vue:10]: There is still the a in the b :)

**Review by spiess-demos — CHANGES_REQUESTED** [review]

File: client/js/components/button/HamburgerMenuButton.vue (hunk 8 above)
Comment (elmasdemos) [reply client/js/components/button/HamburgerMenuButton.vue]: done https://github.com/demos-europe/demosplan-core/pull/1389/commits/7164868f2774114c952225ab2623e37a6aba2cf6

**Review by spiess-demos — APPROVED** [review]
lgtm! (code review)

**Review by spiess-demos — APPROVED** [review]
lgtm! (code review)

**Review by spiess-demos — APPROVED** [review]
lgtm! (code review)

**Pull request merged**